example-crates/tracker_example_cli/src/tracker.rs

```diff
@@ -0,0 +1,266 @@
+use std::{convert::Infallible, fmt::Debug};
+
+use bdk_chain::{
+    bitcoin::Transaction,
+    indexed_tx_graph::{IndexedAdditions, IndexedTxGraph},
+    keychain::{DerivationAdditions, KeychainTxOutIndex},
+    local_chain::{self, LocalChain},
+    tx_graph::CanonicalTx,
+    Anchor, Append, BlockId, ChainOracle, FullTxOut, ObservedAs, PersistBackend,
+};
+use bdk_file_store::{IterError, Store};
+
+use crate::{RemoteChain, RemoteChainChangeSet};
+
+/// Structure for persisting [`Tracker`] data.
+pub type TrackerStore<K, A, B, C> = Store<Tracker<K, A, B>, ChangeSet<K, A, C>>;
+
+pub type LocalTracker<K, A> = Tracker<K, A, LocalChain>;
+pub type LocalTrackerStore<K, A> = TrackerStore<K, A, LocalChain, local_chain::ChangeSet>;
+pub type LocalTrackerChangeSet<K, A> = ChangeSet<K, A, local_chain::ChangeSet>;
+
+pub type RemoteTracker<K, A, O> = Tracker<K, A, RemoteChain<O>>;
+pub type RemoteTrackerStore<K, A, O> = TrackerStore<K, A, RemoteChain<O>, RemoteChainChangeSet>;
+pub type RemoteTrackerChangeSet<K, A> = ChangeSet<K, A, RemoteChainChangeSet>;
```

Comment:
Do we plan to have an example with `RemoteOracle`? It could be interesting to measure performance with and without caching chaindata.